### ISSUE TITLE
Rename gate bug when GatingStrategy has more than 1 gate with same name

### ISSUE THREAD

**Issue by catcd** [whitews/FlowKit#231]
![image](https://github.com/user-attachments/assets/dc076bbf-487c-4cf8-9998-730c1609e8a8)

**Problem Description:**  
There is an issue in the `rename_gate` function at line **259**, causing an inconsistency in the gate hierarchy when renaming a gate. The problem arises during the retrieval of successor gates after the initial renaming process.

**Steps to Reproduce:**

1. Start with the following gate hierarchy:
    ```
    root
    ├── CD4 mémoires
    │   ╰── A
    │       ├── A1
    │       ╰── A2
    ╰── CD4 naifs
        ╰── A
            ├── A1
            ╰── A2
    ```
2. Call the function: `rename_gate('A', 'A new', ('root', 'CD4 naifs'))`.

**Observed Behavior:**

- The renaming process occurs prematurely between **lines 247 and 254**, changing the name `A` to `A new`.
- When attempting to retrieve the successor gate `'A1'` in **line 259** using the hierarchy `('root', 'CD4 naifs', 'A')`, it fails because:
  - There are now **two `A1` gates** in the hierarchy.
  - No gate matches the new name `A new`.

**Potential Root Cause:**

- The renaming logic alters the hierarchy prematurely (between lines **247 and 254**) before successor gates are resolved. This leads to mismatched references when `rename_gate` attempts to access successor gates in **line 259**.

**Suggested Fix:**

- I have fix the problem by add this line before **line 259**:

```
s_tuple = tuple(s if s != gate_name else new_gate_name for s in s_tuple)
```

**Comment by catcd** [comment]
This bug does not happen if these is no A1, A2 in `('root', 'CD4 mémoires', 'A')`, aka A1, A2 has only 1 note match

**Comment by whitews** [comment]
Hi Cat,

Thanks for reporting this. I was able to replicate and have hopefully fixed the issue in the latest `develop` branch. Can you check out the branch and verify it solves the issue for you as well? If it does, I can release a new 1.2.2 version.

Regards,
Scott

**Comment by whitews** [comment]
I've released 1.2.2 with this fix. Please reopen if you still encounter this issue.
